clear_all_faults_main.cpp

```diff
@@ -0,0 +1,159 @@
+
+    properties.append(interface);
+    properties.append(property);
+    properties.append(data);
+
+    try
+    {
+        auto result = bus.call(properties);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        std::cerr << e.what() << std::endl;
+        std::cerr << "\nCall to Set " << property << " property failed for "
+                  << object << std::endl;
+    }
+}
+
+GetPropertyType getDbusProperty(const std::string& service,
+                                const std::string& object,
+                                const std::string& interface,
+                                const std::string& property)
+{
+    auto bus = sdbusplus::bus::new_default();
+    auto properties =
+        bus.new_method_call(service.c_str(), object.c_str(),
+                            "org.freedesktop.DBus.Properties", "Get");
+
+    properties.append(interface);
+    properties.append(property);
+    GetPropertyType value;
+
+    try
+    {
+        auto result = bus.call(properties);
+        result.read(value);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        std::cerr << e.what() << std::endl;
+        std::cerr << "Call to Get " << property << " property failed.";
+    }
+    return value;
+}
+
+int main()
+{
+    std::cout << "\n Starting to clear all fault leds on the system."
+              << std::endl;
+
+    // Check chassis power state
+    GetPropertyType getDbusValue = getDbusProperty(
+        chassisService, chassis0Object, chassisInterface, "CurrentPowerState");
+
+    // If chassis is powered ON, exit without clearing all faults
+    std::string chassisStatus;
+
+    if (auto pVal = std::get_if<std::string>(&getDbusValue))
+    {
+        chassisStatus.assign(pVal->data(), pVal->size());
+    }
+
+    if (chassisStatus == chassisPowerOn)
+    {
+        std::cout << "\nCurrent chassis power state is " << chassisStatus
+                  << ". Exit successfully without doing any fault LED reset."
+                  << std::endl;
+        return 0;
+    }
+
+    std::cout
+        << "\nChassis power state is OFF. Continue to clear all fault leds on the system."
+        << std::endl;
+
+    std::vector<std::string> interfaces = {operationalStatusIntf};
+    GetSubTreePathsResponse inventoryFRUPaths =
+        getObjectSubtreeForInterfaces(inventoryPath, 0, interfaces);
+
+    if (inventoryFRUPaths.size() == 0)
+    {
+        return -1;
+    }
+
+    for (auto fruPath : inventoryFRUPaths)
+    {
+        // Skip CPU Core object paths as it is hosted by PLDM service.
+        if (fruPath.find("core") != std::string::npos)
+        {
+            std::cout << "\nSkipping the object " << fruPath
+                      << " hosted by PLDM service." << std::endl;
+            continue;
+        }
+
+        // Set Functional to true for other inventory paths
+        std::cout << "\nSet functional to true for " << fruPath << std::endl;
```

Comment:
this is probably just for testing, but I'd take this out before it merges.

Reply:
that's right. its just for testing